CRM/Case/ActionMapping.php

```diff
@@ -0,0 +1,138 @@
+<?php
+/*
+ +--------------------------------------------------------------------+
+ | CiviCRM version 5                                                  |
+ +--------------------------------------------------------------------+
+ | Copyright CiviCRM LLC (c) 2004-2019                                |
+ +--------------------------------------------------------------------+
+ | This file is a part of CiviCRM.                                    |
+ |                                                                    |
+ | CiviCRM is free software; you can copy, modify, and distribute it  |
+ | under the terms of the GNU Affero General Public License           |
+ | Version 3, 19 November 2007 and the CiviCRM Licensing Exception.   |
+ |                                                                    |
+ | CiviCRM is distributed in the hope that it will be useful, but     |
+ | WITHOUT ANY WARRANTY; without even the implied warranty of         |
+ | MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.               |
+ | See the GNU Affero General Public License for more details.        |
+ |                                                                    |
+ | You should have received a copy of the GNU Affero General Public   |
+ | License and the CiviCRM Licensing Exception along                  |
+ | with this program; if not, contact CiviCRM LLC                     |
+ | at info[AT]civicrm[DOT]org. If you have questions about the        |
+ | GNU Affero General Public License or the licensing of CiviCRM,     |
+ | see the CiviCRM license FAQ at http://civicrm.org/licensing        |
+ +--------------------------------------------------------------------+
+ */
+
+/**
+ * Class CRM_Event_ActionMapping
+ *
+ * This defines the scheduled-reminder functionality for CiviCase entities.
+ */
+class CRM_Case_ActionMapping extends \Civi\ActionSchedule\Mapping {
+
+  /**
+   * The value for civicrm_action_schedule.mapping_id which identifies the
+   * "Case" mapping.
+   *
+   * Note: This value is chosen to match legacy DB IDs.
+   */
+  const CASE_MAPPING_ID = 99;
+
+  /**
+   * Register Case-related action mappings.
+   *
+   * @param \Civi\ActionSchedule\Event\MappingRegisterEvent $registrations
+   */
+  public static function onRegisterActionMappings(\Civi\ActionSchedule\Event\MappingRegisterEvent $registrations) {
+    $registrations->register(CRM_Case_ActionMapping::create(array(
+      'id' => CRM_Case_ActionMapping::CASE_MAPPING_ID,
+      'entity' => 'civicrm_case',
+      'entity_label' => ts('Case'),
+      'entity_value' => 'case_type',
+      'entity_value_label' => ts('Case Type'),
+      'entity_status' => 'case_status',
+      'entity_status_label' => ts('Case Status'),
+      'entity_date_start' => 'case_start_date',
+      'entity_date_end' => 'case_end_date',
+    )));
+  }
+
+  /**
+   * @inheritdoc
+   */
+  public function getRecipientListing($type) {
```

Comment:
Can we use short array syntax in this function.

Reply:
done